storage_tiering.cpp

```diff
@@ -793,12 +849,13 @@ namespace irods {
                                         config_.group_attribute,
                                         _object_path,
                                         source_replica_number);
-            const auto low_tier_resource_name = get_restage_tier_resource_name(
-                                                    comm_,
-                                                    group_name);
-            // do not queue movement if data is on minimum tier
+            const auto low_tier_resource_name = get_restage_tier_resource_name(comm_, group_name);
+            const auto source_resource_tier = get_resource_group_tier(comm_, _source_resource, group_name);
+            const auto low_tier_resource_tier = get_resource_group_tier(comm_, low_tier_resource_name, group_name);
+
+            // do not queue movement if data is on minimum tier or lower
             // TODO:: query for already queued movement?
```

Comment:
what does this TODO mean to you?

Reply:
I haven't tried this out myself, but I think it may be possible to schedule a bunch of restages by running `iget` repeatedly on the same data object and the TODO reads to me like we need to make sure there isn't already a restage scheduled for that particular source/destination/replica/etc. Thoughts?

Reply:
https://github.com/irods/irods_capability_storage_tiering/issues/103 is the issue that I was thinking about. https://github.com/irods/irods_capability_storage_tiering/issues/108 seems directly related to this TODO. This commit adds a bit of metadata to indicate when an object has already been scheduled for migration: https://github.com/irods/irods_capability_storage_tiering/commit/900307b46b90f65441eaf1d5f0831f2636cb06ef

There is no test for this, so I've created an issue to add a test and remove the TODO comment if resolved: https://github.com/irods/irods_capability_storage_tiering/issues/245